src/UI/Pages/EditInstance/EditInstancePage.test.tsx

```diff
@@ -110,9 +111,59 @@ test("EditInstance View shows success form", async () => {
     body: {
       attributes: {
         bandwidth: "2",
-        circuits: [{}],
       },
     },
     environment: "env",
   });
 });
+
+test("Given the EditInstance View When changing an embedded entity Then the correct request is fired", async () => {
+  const { component, apiHelper } = setup();
+  render(component);
+  const { service_entity, id, version } = ServiceInstance.a;
+
+  expect(
+    await screen.findByRole("generic", { name: "EditInstance-Loading" })
+  ).toBeInTheDocument();
+
+  apiHelper.resolve(Either.right({ data: ServiceInstance.a }));
+
+  expect(
+    await screen.findByRole("generic", { name: "EditInstance-Success" })
+  ).toBeInTheDocument();
+
+  userEvent.click(screen.getByRole("button", { name: "circuits" }));
+  userEvent.click(screen.getByRole("button", { name: "1" }));
+  const serviceIdField = screen.getByText("service_id");
+  userEvent.type(serviceIdField, "{backspace}7");
+
+  const bandwidthField = screen.getByText("bandwidth");
+  expect(bandwidthField).toBeVisible();
+
+  userEvent.type(bandwidthField, "2");
+  userEvent.click(screen.getByText("Confirm"));
+
+  expect(apiHelper.pendingRequests).toHaveLength(1);
+  if (ServiceInstance.a.active_attributes) {
```

Comment:
I am not a fan of this nesting.
We know it is defined.
If it is just to get rid of the type issue, maybe do an early throw or return.
Having this nested seems strange.
But up to you.

Reply:
Yeah, it was indeed a type issue, based on the type it can be `null`, but in this case we know it's not. I'll simplify it